Lib/test/test_urllib2_localnet.py

```diff
@@ -336,8 +337,9 @@ def restore_environ(old_environ):
             os.environ.clear()
             os.environ.update(old_environ)
         self.addCleanup(restore_environ, os.environ.copy())
-        os.environ['NO_PROXY'] = ''
-        os.environ['no_proxy'] = ''
+        self._proxy_bypass = unittest.mock.patch('urllib.request.proxy_bypass', return_value=False)
+        self._proxy_bypass.start()
+        self.addCleanup(self._proxy_bypass.stop)
```

Comment:
The `restore_environ` in L336~L339 should be removed if we don't need to modify environment variables.

Additionally, `proxy_bypass` doesn't need to be an instance variable:

```suggestion
        proxy_bypass = unittest.mock.patch('urllib.request.proxy_bypass', return_value=False)
        proxy_bypass.start()
        self.addCleanup(proxy_bypass.stop)
```

Reply:
Got, update as requested